msu/hooks/entity/world/party.nut

```diff
@@ -332,43 +197,20 @@
 
-		if(::Math.round(moddedSpeed) != ::Math.round(speed))
-		{
-			::MSU.Mod.Debug.printError("Movement Speed for party " + this.getName() + " is NOT correct. Expected: " + speed + " , actual: " + moddedSpeed, "movement")
-			::MSU.Mod.Debug.printError("COMPARING VANILLA AND MSU", "movement")
-			::MSU.Mod.Debug.printError("vanillaMovementSpeed " + vanillaMovementSpeed + " " + this.getBaseMovementSpeedMult() * 100, "movement")
-			::MSU.Mod.Debug.printError("GlobalMult" + GlobalMult + " " + this.getGlobalMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("slowDownPartyPerTroop" + slowDownPartyPerTroop + " " + this.getSlowdownPerUnitMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("NotPlayerMult " + NotPlayerMult + " " + this.getNotPlayerMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("RiverMult " + RiverMult + " " + this.getRiverMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("NighttimeMult " + NighttimeMult + " " + this.getNightTimeMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("TerrainTypeSpeedMult " + TerrainTypeSpeedMult + " " + this.getRoadMovementSpeedMult(), "movement")
-			::MSU.Mod.Debug.printError("getTerrainTypeSpeedMultPlayer " + getTerrainTypeSpeedMultPlayer + " " + this.getRoadMovementSpeedMult(), "movement")
-
-		}
-	}
-	
-	local onSerialize = o.onSerialize;
-	o.onSerialize = function( _out )
-	{
-		this.getFlags().set("VanillaBaseMovementSpeed", this.getVanillaBaseMovementSpeed());
-		this.getFlags().set("BaseMovementSpeedMult", this.getBaseMovementSpeedMult());
-		onSerialize(_out);
-	}
-
-	local onDeserialize = o.onDeserialize;
-	o.onDeserialize = function( _in )
-	{
-		onDeserialize(_in);
-		if (this.getFlags().has("VanillaBaseMovementSpeed"))
-		{
-			this.setVanillaBaseMovementSpeed(this.getFlags().get("VanillaBaseMovementSpeed"));
-		}
-		if (this.getFlags().has("BaseMovementSpeedMult"))
+		local function logIfDifferent(_key, _modded, _vanilla)
 		{
-			this.setBaseMovementSpeedMult(this.getFlags().get("BaseMovementSpeedMult"));
+			if (::Math.round(_modded) != ::Math.round(_vanilla))
+			{
+				::MSU.Mod.Debug.printError(format("%s : Vanilla : %f | Modded: %f", _key, _vanilla, _modded), "movement")
+			}
 		}
-		this.resetBaseMovementSpeed();
 
+		::MSU.Mod.Debug.printError("Movement Speed for party " + this.getName() + " is NOT correct. Expected: " + speed + " , actual: " + _moddedSpeed, "movement")
+		logIfDifferent("GlobalMult", GlobalMult, this.getGlobalMovementSpeedMult());
+		logIfDifferent("slowDownPartyPerTroop", slowDownPartyPerTroop, this.getSlowdownPerUnitMovementSpeedMult());
+		logIfDifferent("NotPlayerMult", NotPlayerMult, this.getNotPlayerMovementSpeedMult());
+		logIfDifferent("RiverMult", RiverMult, this.getRiverMovementSpeedMult());
+		logIfDifferent("NighttimeMult", NighttimeMult, this.getNightTimeMovementSpeedMult());
+		logIfDifferent("TerrainTypeSpeedMult", TerrainTypeSpeedMult, this.getRoadMovementSpeedMult());
```

Comment:
I assume this is a temp thing but you should probably be using popup, might be a system we wanna have incorporated

Reply:
Yeah I guess something like this could be used ina testing module